tools/c7n_mailer/c7n_mailer/slack_delivery.py

```diff
@@ -200,8 +200,7 @@ def send_slack_msg(self, key, message_payload):
                 response.status_code, response.text)
             return
 
-        response_json = response.json()
-        if not response_json['ok']:
+        elif response.text != 'ok':
```

Comment:
that doesn't seem valid, it would be worth backing up and verifying what behavior your attempting to correct? ie. do you have a traceback or issue that your looking to resolve via this?

Reply:
never mind traceback is in #4653

Reply:
Yes, you are right.
Slack API and Webhook work differently.
I pushed changes that look better, test should be adjusted though.

Reply:
> never mind traceback is in #4653... however that issue does not correspond to the current trunk version of mailer which already addressed this. namely not attempting to decode from json if response is not 200.

It will not helm, since the problem caused by different content-type returned by API and Webhooks.
1. API returns JSON https://api.slack.com/messaging/sending#publishing
2. Webhook returns plaintext https://api.slack.com/incoming-webhooks#handling_errors

Thus, if response from webhook **is** 200 you will still fail.